spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/query/StringBasedJdbcQuery.java

```diff
@@ -204,23 +207,47 @@ private void convertAndAddParameter(MapSqlParameterSource parameters, Parameter
 		TypeInformation<?> typeInformation = parameter.getTypeInformation();
 
 		JdbcValue jdbcValue;
-		if (typeInformation.isCollectionLike() && value instanceof Collection<?>) {
+		if (typeInformation.isCollectionLike() //
```

Comment:
We should make our concepts more explicit. Binding a single value is fine. Binding a `List` should become more explicit what we're up to. Binding a list of elements would result in parameter expansion while binding a list of arrays results in tuple expansion. 

I suggest that we refactor the code a bit into a form that we extract methods and prepare the target types (i.e. actual type) so that we have a more explicit flow.

Let me know what you think.

Reply:
I'm happy with that. 
Are you thinking refactoring in the strict sense, i.e. we keep the behaviour as it is with this PR?